shared/fs/common/path-item-action/header.js

```diff
@@ -33,36 +32,48 @@ class Header extends React.PureComponent<Props> {
   }
   render() {
     return (
-      <Kb.Box2 direction="vertical" fullWidth={true} centerChildren={true} style={styles.container}>
-        <PathItemIcon path={this.props.path} size={48} style={styles.pathItemIcon} />
-        {Styles.isMobile && <StaticBreadcrumb pathElements={Types.getPathElements(this.props.path)} />}
-        <Kb.Box2 direction="horizontal" style={styles.nameTextBox}>
-          <CommaSeparatedName
-            type="BodySmallSemibold"
-            name={Types.getPathName(this.props.path)}
-            elementStyle={Styles.collapseStyles([
-              styles.stylesNameText,
-              {color: Constants.getPathTextColor(this.props.path)},
-            ])}
-          />
+      <Kb.Box
+        onClick={
+          // This box is necessary as otherwise the click event propagates into
+          // the ListItem2 backed row.
+          event => event.stopPropagation()
+        }
+      >
+        <Kb.Box2 direction="vertical" fullWidth={true} centerChildren={true} style={styles.container}>
+          <PathItemIcon path={this.props.path} size={48} style={styles.pathItemIcon} />
+          <Kb.WithTooltip
+            containerStyle={styles.nameTextBox}
+            text={Types.pathToString(this.props.path)}
+            multiline={true}
+            showOnPressMobile={true}
+          >
+            <CommaSeparatedName
+              type="BodySmallSemibold"
+              name={Types.getPathName(this.props.path)}
+              elementStyle={Styles.collapseStyles([
+                styles.stylesNameText,
+                {color: Constants.getPathTextColor(this.props.path)},
+              ])}
+            />
+          </Kb.WithTooltip>
+          {this.props.type === 'file' && (
+            <Kb.Text type="BodySmall">{Constants.humanReadableFileSize(this.props.size)}</Kb.Text>
+          )}
+          {this.props.type === 'folder' && (
+            <Kb.Text type="BodySmall">
```

Comment:
I feel like it might be slightly clearer to move this logic out into a function. But it's possible I'm just not used to how many ternary operators are standard in react world.